### PR TITLE
feat(rules): add .claude/rules/ support with path-based matching

### LINKED ISSUE #1504: Docs: Document `require_skills` timing control in rules
## Summary

Document Auto-Claude's `require_skills` extension for rules - the timing control that auto-triggers skills during builds.

## Completed in PR #1396

**Location:** `guides/claude-rules-support.md`

**Covers:**
- `require_skills` frontmatter (Auto-Claude extension, not in Claude Code)
- Timing control: `planning`, `per_subtask`, `end_of_coding`, `qa_phase`
- Optional `paths` filter within require_skills
- Build phase integration diagram
- Links to official Claude Code docs (no duplication)

### PR DESCRIPTION
**Pull request by Quentor1ne**
## Base Branch

- [x] This PR targets the `develop` branch (required for all feature/fix PRs)
- [ ] This PR targets `main` (hotfix only - maintainers)

## Description

Adds `.claude/rules/` support to Auto-Claude - automatically loads project-specific rules based on files being modified during builds.

**Features:**

- Path-based rule matching using glob patterns
- `require_skills` frontmatter extension with timing control (`planning`, `per_subtask`, `end_of_coding`, `qa_phase`)
- Symlink traversal protection
- UI setting to enable/disable rules

**Documentation:**

- Full guide at `guides/claude-rules-support.md`

## Related Issue

Closes #1504

## Type of Change

- [ ] 🐛 Bug fix
- [x] ✨ New feature
- [x] 📚 Documentation
- [ ] ♻️ Refactor
- [x] 🧪 Test

## Area

- [x] Frontend (UI setting)
- [x] Backend (rules loading)

## Commit Message Format

`feat(rules): add .claude/rules/ support with path-based matching`

## Checklist

- [x] I've synced with `develop` branch
- [x] I've tested my changes locally
- [x] I've followed the code principles (SOLID, DRY, KISS)
- [x] My PR is small and focused
- [x] (Python only) All file operations specify `encoding="utf-8"` for text files

## Platform Testing Checklist

- [x] Windows tested (CI)
- [x] macOS tested (CI)
- [x] Linux tested (CI)
- [x] Used centralized `platform/` module instead of direct `process.platform` checks
- [x] No hardcoded paths

## CI/Testing Requirements

- [x] All CI checks pass on **all platforms** (Windows, macOS, Linux)
- [x] All existing tests pass
- [x] New features include test coverage (23 unit tests)

## Screenshots

N/A - Backend feature with minor UI setting addition.

## Feature Toggle

- [x] N/A - Feature is complete and ready for all users

## Breaking Changes

**Breaking:** No

### Release Notes

**New Features**
- Claude Code Rules: load project rules from .claude/rules/ and inject matched rules and required skills into agent system prompts based on file paths and build phase.
- Added useClaudeRules project setting and corresponding environment flag to control rule inclusion (defaults to existing behavior).

**Documentation**
- Added comprehensive Claude Rules Support guide and updated guides index.

**Tests**
- Added unit tests covering rule frontmatter parsing and path-pattern matching.

## Summary by CodeRabbit

* **New Features**
  * Claude Rules support: project rules can be injected into agent prompts with path-based matching and timed require_skills; agent-type aware skill injection.

* **Configuration**
  * New project setting and environment controls to enable/disable Claude Rules (defaults follow existing markdown behavior); explicit on/off supported.

* **Documentation**
  * Added detailed guide describing rule format, setup, and behavior.

* **Tests**
  * Comprehensive unit tests for rule parsing and pattern matching.

* **Chores**
  * Console feedback added for rule loading/matching and injected skills.

### CODE REVIEW

**Comment by Quentor1ne** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Quentor1ne** [comment]
## Feedback Addressed

All CodeRabbit review feedback has been addressed in the latest commits:

**Code fixes (`client.py`):**
- ✅ Moved `import re` and `import fnmatch` to top of file (no inline imports)
- ✅ Fixed indent parsing to use dynamic `skill_base_indent` (supports 2-space YAML)
- ✅ Rewrote glob matching with regex to support multiple `**` patterns
- ✅ Updated skill instruction text to reflect automatic invocation

**Documentation fixes (`claude-rules-support.md`):**
- ✅ Added language specifiers to all fenced code blocks
- ✅ Added blank lines around fenced code blocks (MD031)

**Removed local workaround:**
- ✅ Removed `prebuild-mac.sh` and related `package.json` changes (local fix, not part of feature)

Ready for maintainer review.

**Comment by Quentor1ne** [comment]
## Additional Fixes (93446b10)

Addressed two more issues from CodeRabbit review:

1. **Nested `paths:` list parsing** - Added `in_skill_paths` state to properly handle multi-line `paths:` arrays under structured skills. Previously, nested path list items were incorrectly treated as new skill entries.

2. **Skill de-duplication preserves variants** - Changed deduplication key from just skill name to composite `(skill, when, paths)`. Same skill with different `when` timing or `paths` filters are now preserved separately.

**Comment by Quentor1ne** [comment]
Long story short. I love the idea of Auto Claude. And of course, I started using it.
Only to find out the hard way that everything it created was somehow off, broken, weird, and wrong.
A day later, it was clear, no way to make it follow the rules. :) So now it does. at least for me. Hopefully, it helps everyone. Or if someone has a better idea for solving it... good. Happy to use.

have to admit that skills are so fun and uncharted territory, that it's hard to say how they should actually fit into the picture in the long run. Maybe they need some UI at some point.

Cheers!

**Comment by Quentor1ne** [comment]
Hi @AndyMik90!
Fixed outstanding issues here... seems to be working. 
And it still seems to be a relevant change. 

Cheers!
Ozz

**Review by AndyMik90 — CHANGES_REQUESTED** [review]
## 🤖 Auto Claude PR Review

### Merge Verdict: 🔴 BLOCKED

> **🔴 Blocked** - Merge conflicts must be resolved before merge.

Blocked: PR has merge conflicts with base branch. Resolve conflicts before merge.

### Risk Assessment
| Factor | Level | Notes |
|--------|-------|-------|
| Complexity | High | Based on lines changed |
| Security Impact | Low | Based on security findings |
| Scope Coherence | Good | Based on structural review |

### 🚨 Blocking Issues (Must Fix)
- Merge Conflicts: PR has conflicts with base branch that must be resolved
- CI Failed: CI Complete
- CI Failed: test-frontend (windows-latest)
- CI Failed: test-frontend (ubuntu-latest)
- CI Failed: test-frontend (macos-latest)

### Findings Summary
- **High**: 3 issue(s)
- **Medium**: 2 issue(s)

---
_Generated by Auto Claude PR Review_

### Findings (5 selected of 5 total)

#### 🟡 [4f939fd46e5c] [MEDIUM] [Potential] Symlink traversal in rule file discovery may read files outside project
📁 `apps/backend/core/client.py:762`

The `_collect_all_rules()` function uses `rglob('*.md')` without checking for symlinks. A malicious symlink in `.claude/rules/` pointing outside the project (e.g., to `/etc/passwd` or `~/.ssh/`) would be followed and its content read and included in AI prompts. Other parts of the codebase (apps/backend/agents/utils.py) explicitly skip symlinks for this reason.

**Suggested fix:**
```
Add symlink check before reading: `if rule_path.is_symlink(): logger.warning(f'Skipping symlink: {rule_path}'); continue`
```

#### 🟠 [76754d04df5a] [HIGH] [Potential] Version downgrade from 2.7.5 to 2.7.4 appears unintentional
📁 `apps/frontend/package.json:3`

The frontend package.json has version '2.7.4' but git history shows '2.7.5 release changelog' was already committed. This appears to be an accidental version downgrade during merge/rebase that could cause issues with auto-update mechanisms and semantic versioning expectations.

**Suggested fix:**
```
Update version to 2.7.5 or higher (2.7.6 if adding new features) to maintain proper version progression.
```

#### 🟠 [d0b17fc5dc63] [HIGH] [Potential] Missing test coverage for _parse_rule_frontmatter() - only glob matching tested
📁 `apps/backend/core/test_client.py:1`

The test file only imports and tests `_match_glob_pattern()` (11 test cases). The complex `_parse_rule_frontmatter()` function (185 lines with 5 state variables and ~15 branches) has no test coverage. Critical untested scenarios include: simple vs structured skill formats, nested paths in skills, inline vs multi-line arrays, invalid 'when' values, and edge cases like empty frontmatter.

**Suggested fix:**
```
Add comprehensive tests for _parse_rule_frontmatter() covering: simple paths only, simple skill format, structured skill with when/paths, mixed ordering, invalid values, empty frontmatter, and missing closing ---.
```

#### 🟠 [cf509196e904] [HIGH] [Potential] High cyclomatic complexity in _parse_rule_frontmatter() with 5 state variables
📁 `apps/backend/core/client.py:481`

The function spans ~185 lines with a complex state machine using 5 variables: in_paths, in_skills, in_skill_paths, current_skill, skill_base_indent. The deeply nested conditionals (up to 4 levels) for manual YAML parsing make it difficult to reason about state transitions and prone to edge case bugs.

**Suggested fix:**
```
Consider using PyYAML for frontmatter parsing, or refactor into smaller focused functions: _parse_paths_section(), _parse_skills_section(), _parse_skill_entry(). Extract valid 'when' values to a constant.
```

#### 🟡 [639ea1cd913a] [MEDIUM] [Potential] 446 lines of rules support code should be extracted to separate module
📁 `apps/backend/core/client.py:447`

The rules support section adds ~446 lines (lines 447-893) to client.py, which is already large. This includes 7 functions that form a cohesive unit (should_use_claude_rules, _parse_rule_frontmatter, _match_glob_pattern, _discover_rules_directory, _collect_all_rules, load_claude_rules, _get_files_from_implementation_plan). The codebase has established patterns of extracting features into separate modules (e.g., workspace/ submodule).

**Suggested fix:**
```
Extract to 'core/claude_rules.py' or 'core/rules.py'. Import load_claude_rules() and should_use_claude_rules() into client.py similar to how simple_client.py works.
```

---
*This review was generated by Auto Claude.*

**Comment by Quentor1ne** [comment]
Ready for review! 🎉

This PR adds support for Claude Code's .claude/rules/ convention, enabling path-based rules that Auto-Claude automatically loads based on which files are being modified.

Key features:

Path-based rule matching using glob patterns (src/api/**/*.ts)
require_skills with timing control (planning, per_subtask, end_of_coding, qa_phase)
Defence-in-depth security against symlink traversal attacks
Comprehensive test coverage (23 test cases)
New documentation in guides/claude-rules-support.md
All review feedback addressed:

✅ Symlink security hardened at both the directory and file levels
✅ Exception handling for filesystem errors (OSError, RuntimeError)
✅ Documentation defaults clarified
✅ Tests moved to proper conftest.py pattern
✅ All CodeRabbit threads resolved

Thanks, @AndyMik90, for the thorough reviews. Tough to get is all checking out. But fun challenge. For me, it all started the day after I found Auto Claude and thought, "Well, now life is good!" 😊 Overnight, I let Auto-claude do 10+ tasks and was superhappy - only to find out that the resulting code was a largely unfixable mess. While debugging, I asked Claude to figure out why it happened, and the answer was simple - "I did not follow any of the rules in your project, but I can help you change that"... so here we are... I hope it helps. It did help me...

Keep up the good work!

😀
Ozz

**Review by AndyMik90 — CHANGES_REQUESTED** [review]
## 🤖 Auto Claude PR Review

## 🟠 Follow-up Review: Needs Revision

> **🟠 Needs revision** - 2 unresolved finding(s) from previous review.

### Resolution Status
- ✅ **Resolved**: 3 previous findings addressed
- ❌ **Unresolved**: 2 previous findings remain
- 🆕 **New Issues**: 6 new findings in recent changes

### Finding Validation
- 🔍 **Dismissed as False Positives**: 0 findings were re-investigated and found to be incorrect
- ✓ **Confirmed Valid**: 4 findings verified as genuine issues
- 👤 **Needs Human Review**: 1 findings require manual verification

### 🚨 Blocking Issues
- Branch Out of Date: PR branch is behind the base branch and needs to be updated
- quality: [UNRESOLVED] [Potential] High cyclomatic complexity in _parse_rule_frontmatter() with 5 state variables
- quality: [UNRESOLVED] [Potential] 446 lines of rules support code should be extracted to separate module
- test: test_no_frontmatter assertion contradicts function return value
- test: test_missing_closing_delimiter assertion contradicts function return value
- quality: Skill-level path filters stored but not programmatically applied
- test: [FROM COMMENTS] test_no_frontmatter assertion contradicts function return value
- test: [FROM COMMENTS] test_missing_closing_delimiter assertion contradicts function return value
- quality: [FROM COMMENTS] Skill-level path filters not programmatically applied

### Verdict
All 21 CI checks are now passing (previously had failures), and merge conflicts appear resolved. 3 of 5 previous findings are verified resolved (symlink traversal fix, version restored to 2.7.5, comprehensive test coverage added). However, 2 confirmed test bugs block merge: (1) CMT-001 and CMT-002 - test assertions for test_no_frontmatter and test_missing_closing_delimiter contradict the actual function behavior (_parse_rule_frontmatter returns original content but tests assert empty string). These are medium-severity issues that should be quick fixes. Additionally, CMT-003 (skill-level path filtering) needs human clarification on design intent. The 2 remaining original findings (complexity, module extraction) are valid code quality suggestions but are low-priority and don't block merge given the added test coverage.

### Review Process
Agents invoked: resolution-verifier, comment-analyzer, finding-validator

---
*This is an AI-generated follow-up review using parallel specialist analysis with finding validation.*


### Findings (8 selected of 8 total)

#### 🟠 [cf509196e904] [HIGH] [UNRESOLVED] [Potential] High cyclomatic complexity in _parse_rule_frontmatter() with 5 state variables
📁 `apps/backend/core/client.py:481`

The function spans ~185 lines with a complex state machine using 5 variables: in_paths, in_skills, in_skill_paths, current_skill, skill_base_indent. The deeply nested conditionals (up to 4 levels) for manual YAML parsing make it difficult to reason about state transitions and prone to edge case bugs.

Resolution note: _parse_rule_frontmatter() spans lines 481-666 (~185 lines) with 5 state variables (in_paths, in_skills, in_skill_paths, current_skill, skill_base_indent) and deeply nested conditionals up to 4 levels.

**Suggested fix:**
```
Consider using PyYAML for frontmatter parsing, or refactor into smaller focused functions: _parse_paths_section(), _parse_skills_section(), _parse_skill_entry(). Extract valid 'when' values to a constant.
```

#### 🟡 [639ea1cd913a] [MEDIUM] [UNRESOLVED] [Potential] 446 lines of rules support code should be extracted to separate module
📁 `apps/backend/core/client.py:447`

The rules support section adds ~446 lines (lines 447-893) to client.py, which is already large. This includes 7 functions that form a cohesive unit (should_use_claude_rules, _parse_rule_frontmatter, _match_glob_pattern, _discover_rules_directory, _collect_all_rules, load_claude_rules, _get_files_from_implementation_plan). The codebase has established patterns of extracting features into separate modules (e.g., workspace/ submodule).

Resolution note: Rules support section spans lines 447-891 (~445 lines, 7 functions) within client.py which totals 1456 lines.

**Suggested fix:**
```
Extract to 'core/claude_rules.py' or 'core/rules.py'. Import load_claude_rules() and should_use_claude_rules() into client.py similar to how simple_client.py works.
```

#### 🟡 [CMT-001] [MEDIUM] test_no_frontmatter assertion contradicts function return value
📁 `apps/backend/core/test_client.py:222`

The test asserts rule_content == '' but _parse_rule_frontmatter() at client.py:514-515 returns ([], [], content) where content is the original non-empty markdown string. The test will fail when executed. Either the function should return empty string for no-frontmatter content, or the test assertion should match the actual return value.

**Suggested fix:**
```
Either change client.py line 515 to 'return [], [], ""' or change test assertion to 'assert rule_content == content.strip()'. The test comment suggests returning empty is the intended behavior.
```

#### 🟡 [CMT-002] [MEDIUM] test_missing_closing_delimiter assertion contradicts function return value
📁 `apps/backend/core/test_client.py:240`

Same issue as CMT-001. The function returns ([], [], content) at client.py:525-526 for missing closing delimiter, but the test asserts rule_content == ''. These are contradictory.

**Suggested fix:**
```
Align test assertion with actual function behavior, or modify function to return empty string for malformed frontmatter cases.
```

#### 🟡 [CMT-003] [MEDIUM] Skill-level path filters stored but not programmatically applied
📁 `apps/backend/core/client.py:876`

When a rule matches, all skills are added via _add_skill() without checking the skill's own 'paths' filter against files_to_check. The paths data is preserved and rendered as advisory text in the agent prompt but not used for programmatic filtering. A skill scoped to 'src/api/**' would be included even when only 'src/components/**' files are modified.

**Suggested fix:**
```
Before calling _add_skill(skill), check if skill has 'paths' key and if so, verify at least one path matches files_to_check. Or document that skill-level paths are advisory only.
```

#### 🟡 [CMT-001] [MEDIUM] [FROM COMMENTS] test_no_frontmatter assertion contradicts function return value
📁 `apps/backend/core/test_client.py:222`

Test asserts rule_content == '' but function returns original content string when no frontmatter found.

**Suggested fix:**
```
Change function to return '' or fix test to match actual return value.
```

#### 🟡 [CMT-002] [MEDIUM] [FROM COMMENTS] test_missing_closing_delimiter assertion contradicts function return value
📁 `apps/backend/core/test_client.py:240`

Same mismatch as CMT-001 for the missing closing delimiter case.

#### 🟡 [CMT-003] [MEDIUM] [FROM COMMENTS] Skill-level path filters not programmatically applied
📁 `apps/backend/core/client.py:876`

Skills with paths filters are added unconditionally; paths are advisory text in prompt only.

---
*This review was generated by Auto Claude.*

File: apps/backend/core/client.py [hunk 1]
```diff
@@ -442,6 +442,10 @@ def load_claude_md(project_dir: Path) -> str | None:
     return None
 
 
+# Note: Claude Code Rules support is in core/claude_rules.py
+# Feature flags (should_use_claude_md, should_use_claude_rules) are in core/config.py
+
+
 def create_client(
```
[the remark this answers is not in the document]
Comment (Quentor1ne) [reply apps/backend/core/client.py:448]: Bug confirmed and fixed.

**Comment by Quentor1ne** [comment]
## 🤖 PR Response: `.claude/rules/` Support

### ✅ Resolution Status

| Finding | Status | Resolution |
|---------|--------|------------|
| [HIGH] Cyclomatic complexity in `_parse_rule_frontmatter()` | ✅ Resolved | Refactored into 4 focused helper functions |
| [MEDIUM] 446 lines should be extracted to separate module | ✅ Resolved | Extracted to `core/claude_rules.py` |
| [CMT-001] `test_no_frontmatter` assertion mismatch | ✅ Resolved | Function now returns empty string for invalid frontmatter |
| [CMT-002] `test_missing_closing_delimiter` assertion mismatch | ✅ Resolved | Function now returns empty string for invalid frontmatter |
| [CMT-003] Skill-level path filters not applied | ✅ Resolved | Implemented skill-level path filtering in `load_claude_rules()` |

---

### 📦 Changes in This PR

#### 1. New Module: `core/claude_rules.py`
- Extracted ~480 lines from `client.py` into dedicated module
- Custom lightweight YAML frontmatter parser (intentionally not PyYAML to avoid dependencies)
- Supports: `paths` lists, `require_skills` with nested properties, inline arrays `[a, b]`
- Does NOT support: anchors, aliases, multi-line strings (`|` or `>`)

#### 2. Refactored Parser Functions
Reduced complexity by splitting `_parse_rule_frontmatter()` into focused helpers:
- `_parse_inline_array()` - handles `[item1, item2]` format
- `_parse_paths_section()` - extracts path patterns from frontmatter
- `_parse_skill_entry()` - parses individual skill with `when`/`paths`
- `_parse_skills_section()` - handles full `require_skills:` block

#### 3. Thread-Safe Caching
- 5-minute TTL cache (`_RULES_CACHE`) avoids re-parsing on every `create_client()` call
- `threading.Lock()` for safe concurrent access by parallel agents
- Copy-on-return prevents cache mutation

#### 4. `require_skills` Extension
Auto-Claude extension to Claude Code's rules format:
- `when` timing: `planning`, `per_subtask`, `end_of_coding`, `qa_phase`
- Optional nested `paths` filter for skill-specific file matching
- Skills are now filtered by their own `paths` patterns (CMT-003 fix)

#### 5. Security Hardening
- Symlink traversal protection in `_discover_rules_directory()` and `_collect_all_rules()`
- Path resolution validates rules stay within project directory
- Catches `OSError`, `ValueError`, `RuntimeError` during path resolution

#### 6. Bug Fix: Followup Planner
- `run_followup_planner()` was missing `agent_type="planner"` parameter
- Caused skills with `when: "planning"` to be filtered out (defaulted to "coder")

#### 7. Documentation
- `guides/claude-rules-support.md` with usage examples and build phase diagram
- YAML parsing limitations documented in module docstring

---

### 🔍 Finding Responses

**[HIGH] Cyclomatic complexity** - The original 185-line function with 5 state variables has been decomposed into 4 single-purpose helper functions. Each function now handles one aspect of parsing (inline arrays, paths section, skill entries, skills section).

**[MEDIUM] Module extraction** - All 7 rules-related functions are now in `core/claude_rules.py`. Only `load_claude_rules()` is imported into `client.py`.

**[CMT-001/002] Test assertion mismatches** - `_parse_rule_frontmatter()` now returns `("", [], "")` for content without valid frontmatter (no opening `---` or missing closing delimiter). Tests pass.

**[CMT-003] Skill-level paths not applied** - `load_claude_rules()` now checks skill `paths` against `files_to_check` before including the skill. A skill scoped to `src/api/**` will only be included when API files are being modified.

**Comment by Quentor1ne** [comment]
# Ready when you are
@AndyMik90 Thanks! I think this now is pretty good :)

 ## Overview: Rules Module Refactoring & Caching Validation                                  
                                                                                              
  ### What Changed                                                                            
                                                                                              
  The rules loading code was extracted from `client.py` into a dedicated                      
  `core/claude_rules.py` module with a 5-minute TTL cache.                                    
                                                                                              
  **Before (in client.py):**                                                                  
  - Rules parsed on every `create_client()` call                                              
  - No caching - 60+ file reads per task                                                      
  - Code mixed with client configuration logic                                                
                                                                                              
  **After (separate claude_rules.py):**                                                       
  - Dedicated module with single responsibility                                               
  - Thread-safe cache with 300-second TTL                                                     
  - Rules parsed once, served from memory on subsequent calls                                 
                                                                                              
  ---                                                                                         
                                                                                              
  ### Real-World Validation                                                                   
                                                                                              
  **Test task:** 8 subtasks across 6 phases, 20 rule files in `.claude/rules/`                
                                                                                              
  **Observed Behavior:**                                                                      
                                                                                              
  1. **Rules ARE being followed** - Implementation plan and code showed:                      
     - `sanitizeForPostgrestFilter()` used (from `security/api-routes.md`)                    
     - 300ms debounce implemented (from `frontend/patterns.md`)                               
     - Offset pagination patterns applied (from `performance/queries.md`)                     
                                                                                              
  2. **Cache instrumentation confirmed:**                                                     
     Call 1: MISS #1 - parsing 19 rules from disk                                             
     Call 2: HIT #1 (age: 0.0s) - returning 19 cached                                         
     Call 3: HIT #2 (age: 0.0s) - returning 19 cached                                         
     Call 4: HIT #3 (age: 0.0s) - returning 19 cached                                         
                                                                                              
  3. **Performance impact:**                                                                  
  | Metric | Without Cache | With Cache |                                                     
  |--------|---------------|------------|                                                     
  | File reads (4 calls) | 76 | 19 |                                                          
  | File reads (8 subtasks) | 152+ | 19 |                                                     
  | File reads (60 calls) | 1140+ | 19-38 |                                                   
                                                                                              
  ---                                                                                         
                                                                                              
  ### Key Findings                                                                            
                                                                                              
  1. **Cache works correctly** - Verified with direct testing showing 1 MISS followed by HITs 
                                                                                              
  2. **Thread safety confirmed** - Uses `threading.Lock()` for concurrent access protection   
                                                                                              
  3. **TTL appropriate** - 5-minute cache covers typical task duration (subtasks completed in 
  ~10 min)                                                                                    
                                                                                              
  4. **Rules content validated** - Agent incorporated rule patterns into implementation plan  
  and generated code                                                                          
                                                                                              
  ---                                                                                         
                                                                                              
  ### Files Modified                                                                          
                                                                                              
  - `apps/backend/core/claude_rules.py` - New dedicated module with caching                   
  - `apps/backend/core/client.py` - Imports from claude_rules.py                              
  - `apps/backend/core/config.py` - Feature flags (`should_use_claude_rules()`)               
                                                                                              
  ---                                                                                         
                                                                                              
  ### Conclusion                                                                              
                                                                                              
  The refactoring successfully:                                                               
  - ✅ Reduced file I/O by ~98% (1-2 parses vs 60+)                                           
  - ✅ Maintained correct rule matching behavior                                              
  - ✅ Preserved thread safety for parallel agents                                            
  - ✅ Rules content verified in real task output

**Comment by AndyMik90** [comment]
👋 Thanks so much for your interest in contributing to Aperant!

**Pull requests are currently paused.** We're rebuilding the entire app from the ground up for **Aperant 3.0** (including new cloud features). Most of that work is happening in a separate development repo right now and will be merged here once it's ready — so although this repo looks quiet, the project is very much alive. 🚀

Because the current codebase is being fully replaced, we're closing existing PRs so contributors aren't left waiting on changes that unfortunately can't be merged. This is **not** a reflection on your work, and we're sorry for any wasted effort. 🙏

We'd genuinely love your help shaping 3.0 instead:

- 🐛 **Report a bug:** https://github.com/AndyMik90/Aperant/issues
- 💡 **Share an idea / feature request:** https://github.com/AndyMik90/Aperant/discussions
- 💬 **Join the Discord:** https://discord.gg/KCXaPBr4Dj

Contributions will reopen once the 3.0 foundation lands. Thank you for understanding!
